force-app/main/default/triggers/AccountTrigger.trigger

```diff
@@ -0,0 +1,36 @@
+trigger AccountTrigger on Account (before insert,after insert) {
+    if (trigger.isBefore) {
+        for (Account acc : Trigger.new) {
+            if (acc.Type == null) {
+                acc.Type = 'Prospect';
+            }
+         
+            if (acc.ShippingStreet != null || acc.ShippingCity != null || acc.ShippingState != null || acc.ShippingPostalCode != null || acc.ShippingCountry!= null) {
+                acc.BillingStreet = acc.ShippingStreet;
+                acc.BillingCity = acc.ShippingCity;
+                acc.BillingState = acc.ShippingState;
+                acc.BillingPostalCode = acc.ShippingPostalCode;
+                acc.BillingCountry = acc.ShippingCountry;
+            }
+     
+        if (acc.Phone != null && acc.Website != null && acc.Fax !=null) {
+            acc.rating = 'Hot';
+        }       
+    }
+    }
+       
+    List <Contact> newContacts = new List<Contact>();
+    if (trigger.isAfter) {
+       if (trigger.isInsert) {
+        for (Account acc : trigger.new) {
+            Contact con = new Contact();
+            con.AccountId = acc.Id;
+            con.LastName = 'DefaultContact';
+            con.Email = 'default@email.com';
+            newContacts.add(con);
+        }
+        insert newContacts;
```

Comment:
This is not clear in the requirements, but you probably want to check that there is not a contact with these values already. Otherwise, with no other criteria, it will create a new contact on every update. Probably not the desired behavior, and this would be something you ask for clarification on as a developer.